projects/optic/src/commands/capture/storage.ts

```diff
@@ -0,0 +1,104 @@
+import crypto from 'crypto';
+import fs from 'node:fs/promises';
+import os from 'os';
+import path from 'path';
+import { HttpArchive } from '../oas/captures';
+import { OperationQueries } from '../oas/operations/queries';
+import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';
+import { getEndpointId } from '../../utils/id';
+
+const tmpDirectory = os.tmpdir();
+
+export const UNMATCHED_PATH = 'unmatched';
+
+export async function getCaptureStorage(filePath: string): Promise<string> {
+  const resolvedFilepath = path.resolve(filePath);
+
+  const specPathHash = crypto
+    .createHash('md5')
+    .update(resolvedFilepath)
+    .digest('hex');
+
+  // TODO in the future we can reuse cached requests if paths are identical (maybe hash paths?)
+  const trafficDirectory = path.join(
+    tmpDirectory,
+    'optic',
+    'captures-v2',
+    specPathHash,
+    String(Date.now())
+  );
+
+  await fs.mkdir(trafficDirectory, { recursive: true });
+
+  return trafficDirectory;
+}
+
+export class GroupedCaptures {
+  private paths: Map<string, { path: string; hars: HttpArchive.Entry[] }>;
+  private queries: OperationQueries;
+  constructor(
+    trafficDir: string,
+    endpoints: { path: string; method: string }[]
+  ) {
+    this.queries = new OperationQueries(
+      endpoints.map((e) => ({
+        pathPattern: e.path,
+        method: e.method,
+        specPath: jsonPointerHelpers.compile(['paths', e.path, e.method]),
+      }))
+    );
+    this.paths = new Map();
+    for (const endpoint of endpoints) {
+      const id = getEndpointId(endpoint);
+      const endpointHash = crypto.createHash('md5').update(id).digest('hex');
+
+      const fPath = path.join(trafficDir, `${endpointHash}.har`);
+
+      this.paths.set(id, {
+        path: fPath,
+        hars: [],
+      });
+    }
+
+    const unmatched = path.join(trafficDir, `unmatched.har`);
+    this.paths.set(UNMATCHED_PATH, {
+      path: unmatched,
+      hars: [],
+    });
+  }
+
+  addHar(har: HttpArchive.Entry) {
```

Comment:
Collects this to be synchronous - likely we don't need streams here as it's likely we're going to get requests in the range of 100-1000 (maybe)